In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
import seaborn
seaborn.set()

In [3]:
data_f=pd.read_csv('/content/1.02.+Multiple+linear+regression.csv')

In [4]:
data_f

SAT   GPA  Rand 1,2,3
0   1714  2.40           1
1   1664  2.52           3
2   1760  2.54           3
3   1685  2.74           3
4   1693  2.83           2
..   ...   ...         ...
79  1936  3.71           3
80  1810  3.71           1
81  1987  3.73           3
82  1962  3.76           1
83  2050  3.81           2

[84 rows x 3 columns]

GPA=BO +B1SAT +B2Rand1,2,3

In [5]:
data_f.describe()

SAT        GPA  Rand 1,2,3
count    84.000000  84.000000   84.000000
mean   1845.273810   3.330238    2.059524
std     104.530661   0.271617    0.855192
min    1634.000000   2.400000    1.000000
25%    1772.000000   3.190000    1.000000
50%    1846.000000   3.380000    2.000000
75%    1934.000000   3.502500    3.000000
max    2050.000000   3.810000    3.000000

In [10]:
y=data_f['GPA']
x1=data_f[['SAT','Rand 1,2,3']]

In [11]:
x=sm.add_constant(x1)
results = sm.OLS(y,x).fit()

In [13]:
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    GPA   R-squared:                       0.407
Model:                            OLS   Adj. R-squared:                  0.392
Method:                 Least Squares   F-statistic:                     27.76
Date:                Thu, 18 May 2023   Prob (F-statistic):           6.58e-10
Time:                        04:23:46   Log-Likelihood:                 12.720
No. Observations:                  84   AIC:                            -19.44
Df Residuals:                      81   BIC:                            -12.15
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.2960      0.417      0.710      0.480      -0.533       1.125
SAT            0.0017      0.000      7.432      0.000       0.001       0.002
Rand 1,2,3    -0.0083      0.027     -0.304      0.762      -0.062       0.046
==============================================================================
Omnibus:                       12.992   Durbin-Watson:                   0.948
Prob(Omnibus):                  0.002   Jarque-Bera (JB):               16.364
Skew:                          -0.731   Prob(JB):                     0.000280
Kurtosis:                       4.594   Cond. No.                     3.33e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.33e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

y^2=0.296 + 0.0017x1 - 0.0083x2

# Multiple Linear Regression - Exercise

You are given a real estate dataset. 

Real estate is one of those examples that every regression course goes through as it is extremely easy to understand and there is a (almost always) certain causal relationship to be found.

The data is located in the file: 'real_estate_price_size_year.csv'. 

You are expected to create a multiple linear regression (similar to the one in the lecture), using the new data. 

In this exercise, the dependent variable is 'price', while the independent variables are 'size' and 'year'.

Good luck!

## Import the relevant libraries

In [14]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
import seaborn
seaborn.set()

## Load the data


In [15]:
pd_data=pd.read_csv('/content/real_estate_price_size_year.csv')

In [16]:
pd_data

price     size  year
0   234314.144   643.09  2015
1   228581.528   656.22  2009
2   281626.336   487.29  2018
3   401255.608  1504.75  2015
4   458674.256  1275.46  2009
..         ...      ...   ...
95  252460.400   549.80  2009
96  310522.592  1037.44  2009
97  383635.568  1504.75  2006
98  225145.248   648.29  2015
99  274922.856   705.29  2006

[100 rows x 3 columns]

In [20]:
pd_data.describe()

price         size         year
count     100.000000   100.000000   100.000000
mean   292289.470160   853.024200  2012.600000
std     77051.727525   297.941951     4.729021
min    154282.128000   479.750000  2006.000000
25%    234280.148000   643.330000  2009.000000
50%    280590.716000   696.405000  2015.000000
75%    335723.696000  1029.322500  2018.000000
max    500681.128000  1842.510000  2018.000000

## Create the regression


### Declare the dependent and the independent variables


In [17]:
y=pd_data['price']
x1=pd_data[['size','year']]

### Regression


In [18]:
x=sm.add_constant(x1)
results=sm.OLS(y,x).fit()

In [19]:
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  price   R-squared:                       0.776
Model:                            OLS   Adj. R-squared:                  0.772
Method:                 Least Squares   F-statistic:                     168.5
Date:                Thu, 18 May 2023   Prob (F-statistic):           2.77e-32
Time:                        04:37:39   Log-Likelihood:                -1191.7
No. Observations:                 100   AIC:                             2389.
Df Residuals:                      97   BIC:                             2397.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -5.772e+06   1.58e+06     -3.647      0.000   -8.91e+06   -2.63e+06
size         227.7009     12.474     18.254      0.000     202.943     252.458
year        2916.7853    785.896      3.711      0.000    1357.000    4476.571
==============================================================================
Omnibus:                       10.083   Durbin-Watson:                   2.250
Prob(Omnibus):                  0.006   Jarque-Bera (JB):                3.678
Skew:                           0.095   Prob(JB):                        0.159
Kurtosis:                       2.080   Cond. No.                     9.41e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 9.41e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""